### PR TITLE
OCPBUGSM-29119 Can't change Installation disk

### PR DESCRIPTION
**Pull request by michaellevy101**
When patching a cluster first we update the hosts according to the parameters (For instance updating the installation path which succeed and updates the DB)
Then calling to the inventory.updateHostsAndClusterStatus which reads a new cluster object from the DB but calling the actual update with the old host objects (contain the old installation path in our example)
The updateInventory just reverts the user selection and reselect the old installationDiskId/Path
We have another issue that changing the selected disk add more and more identical messages.

### CODE REVIEW

File: internal/hardware/validator.go [hunk 1]
```diff
@@ -273,8 +273,8 @@ func (v *validator) getOCPRequirementsForVersion(cluster *common.Cluster) (*mode
 	return requirements, nil
 }
 
-func compileDiskReasonTemplate(template string, wildcards ...string) *regexp.Regexp {
-	tmp, err := regexp.Compile(fmt.Sprintf(regexp.QuoteMeta(template), wildcards))
+func compileDiskReasonTemplate(template string, wildcards ...interface{}) *regexp.Regexp {
```
Comment (jakub-dzon) [inline internal/hardware/validator.go:276]: Why was this change needed? After all the wildcards will always be strings.

Comment (michaellevy101) [reply internal/hardware/validator.go:276]: The compileDiskReasonTemplate is a string variadic function while Sprintf is an Interface variadic function.
We must pass the exact type to the function.
In addition we must expand this array before passing it to the Sprintf method, else Sprintf method would deal with it as a single argument instead of an array of arguments.
The easy path is just changing the slice type to an interface slice and eliminate the explicit conversion (Creating a new slice and copy all the values)

Comment (michaellevy101) [reply internal/hardware/validator.go:276]: In logical terms the result pattern was "Disk is too small /(disk only has [.* .*], but %s(Missing) are required/)" instead of "Disk is too small /(disk only has .* , but .* are required/)"

**Review by jakub-dzon — APPROVED** [review]
